File: Python/basic13.py
#1. Print 1-255
def printTo255():
	num = 1
	while num<=255:
		print num
		num +=1

#2. Print 0-255 with sum
def printToSum():
	num = 0
	total = 0
	while num<=255:
		print "Number:{} Sum:{}".format(num, total)
		num +=1
		total += num

#3. Print odds 1-255
def printOdds():
	for i in range (0, 257):
		if i%2 == 1:
			print i

#4. Iterate through array.
def iterate(arr):
	for val in arr:
		print val

#5. Find max in array.
def findMax(arr):
	return max(arr)

#6. Given an array of intergers find average.
def findAvg(arr):
	sum = 0
	for val in arr:
		sum += val
	return sum/len(arr)

#7. Create an array with odd integers between 1 and given number.
def oddArray(n):
	arr = []
	for i in range(1, n+1):
		if i%2==1:
			arr.append(i)
	return arr

#8. Square each value in a given array.
def squareArray(arr):
	for i in range (0, len(arr)):
		arr[i] = arr[i]*arr[i]
	return arr

#9. Given an array and a value y, count and print the number of values greater than y.
def greaterThanY(arr, y):
	count = 0
	for val in arr:
		if val > y:
			count +=1
			print val
	print "Total number of values greater than {}: {}".format(y, count)

#10. Return a given array after setting negative values to zero.
def noNegs(arr):
	for i in range(0, len(arr)):
		if arr[i] < 0:
			arr[i]=0
	return arr

#11. Given an array return the max, min, average.
def maxMinAvg(arr):
	sum = 0
	for val in arr:
		sum += val
	avg = sum/len(arr)
	return "Max:{} Min:{} Avg:{}".format(max(arr), min(arr), avg)

#12. Given an array move all values forward by one, dropping the first and leaving a 0.
def shiftArr(arr):
	for i in range(0, len(arr)-1):
		arr[i] = arr[i+1]
	arr[len(arr)-1]=0
	return arr

#13. Replace any negative numbers in array with string 'hello'.
def swapString(arr):
	for i in range(0, len(arr)):
		if arr[i] < 0:
			arr[i]='hello'
	return arr

print (swapString([1, 4, -5]))

File: Ruby/basic13.rb
#1. Print 1-n
def printInt(n)
	1.upto(n) { |i| puts i}
end

#2. Print odd numbers between 1-255.
def oddNum
	0.upto(255) { |i| puts i unless i.even? }
end


#3. Print sum of all numbers 1-255.
def findSum
	sum = 0
	1.upto(255) { |i| puts sum += i}
end

# findSum
# 4. Iterating through an array and print each value.
def iterate(arr)
	for i in 0...arr.length
		puts arr[i]
	end
end

# 5.Find Max
def findMax(arr)
	puts arr.max
end
# findMax([-2, -5, -6, -8, -10])

#6. Get Average
def getAverage(arr)
	sum = 0
	arr.each { |i| sum += i }
	puts "The average is #{sum/arr.length}"
end
# getAverage([2, 10, 3])
#returns 5

#7. Create an array with only odd numbers 1-255
def createArr
	y = (1..255).to_a.reject {|elem| elem.even? }
	puts y
end
# createArr()

# 8. Greater than y
def greaterThanY(arr, val)
	count = 0
	for i in 0...arr.length
		if arr[i] > val
			count +=1
		end
	end
	puts count
end
# greaterThanY([1, 4, 6, 8, 9], 0)

# 9. Square the values
def squareValues (arr)
	puts arr.map {|i| i**2}
end
squareValues([1, 2, 3, 4])

#10.  Eliminate Negative Numbers--replace with 0.
def noNegs(arr)
	puts arr
	for i in 0...arr.length
		if arr[i] < 0
			arr[i] = 0
		end
	end
	puts arr
end
# noNegs([1, 5, -7, -10, 1])

#11. Max, Min and Average
def maxMinAvg(arr)
	sum = 0
	for i in 0...arr.length
		sum = sum + arr[i]
	end
	puts sum
	puts "The average is #{sum/arr.length}"
	puts "The min is #{arr.min}"
	puts "The max is #{arr.max}"
end
# maxMinAvg([6, 4])

#12. Shifting
def shiftArr(arr)
	print arr
	arr.shift
	arr.push(0)
	print arr
end
# shiftArr([3, 1, 7, 9, 10])

#13. Number to String
def numtoString(arr)
	puts arr
	for i in 0...arr.length
		if arr[i] < 0
			arr[i] = "Dojo"
		end
	end
	puts arr
end
# numtoString([1, 5, -7, -10, 1])